src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs

```diff
@@ -59,14 +66,26 @@ private T RoundTrip<T>(T value, Action<ObjectWriter, T> writeAction, Func<Object
         {
             using var stream = new MemoryStream();
 
-            using (var writer = new ObjectWriter(stream, leaveOpen: true))
+            ImmutableArray<object> keepAliveObjects;
+            using (var writer = new ObjectWriter(stream, leaveOpen: true, canKeepObjectsAlive: true))
             {
                 writeAction(writer, value);
+
+                keepAliveObjects = writer.TakeKeepAliveObjects();
             }
 
             stream.Position = 0;
-            using var reader = ObjectReader.TryGetReader(stream);
-            return readAction(reader);
+
+            try
+            {
+                using var reader = ObjectReader.TryGetReader(stream);
+                return readAction(reader);
+            }
+            finally
+            {
+                foreach (var obj in keepAliveObjects)
+                    GC.KeepAlive(obj);
```

Comment:
don't get the finally need here.

Reply:
I need the `KeepAlive` call after the `return`.

Reply:
➡️ Rewritten to avoid the `finally`